node/src/actors/chain_manager/transaction_factory.rs

```diff
@@ -107,14 +113,20 @@ pub fn insert_change_output(
 }
 
 /// Build value transfer transaction with the given outputs and fee.
-pub fn build_vtt<S: std::hash::BuildHasher>(
+#[allow(clippy::too_many_arguments)]
+pub fn build_vtt(
     outputs: Vec<ValueTransferOutput>,
     fee: u64,
-    own_utxos: &mut HashMap<OutputPointer, u64, S>,
+    own_utxos: &mut OwnUnspentOutputsPool,
     own_pkh: PublicKeyHash,
     all_utxos: &UnspentOutputsPool,
     timestamp: u64,
     tx_pending_timeout: u64,
+    // Sorted utxo by value
+    // None: no sorted
+    // Some(true): bigger value first
+    // Some(false): lower value first
+    sorted_bigger: Option<bool>,
```

Comment:
This should be an enum:

```rust
enum UtxoSelectionStrategy {
    None, // or Random
    BigFirst,
    SmallFirst,
}
```

```suggestion
    utxo_selection_strategy: UtxoSelectionStrategy,
```